app/models/manageiq/providers/azure_stack/inventory/parser/network_manager.rb

```diff
@@ -3,11 +3,59 @@ def parse
     log_header = "Collecting data for EMS : [#{collector.manager.name}] id: [#{collector.manager.id}]"
     $azure_stack_log.info("#{log_header}...")
 
+    cloud_networks
+    network_ports
     security_groups
 
     $azure_stack_log.info("#{log_header}...Complete")
   end
 
+  def cloud_networks
+    collector.networks.each do |network|
+      persister.cloud_networks.build(
+        :name    => network.name,
+        :ems_ref => network.id.downcase
+      )
+      if network.subnets
+        cloud_subnets(network)
+      end
```

Comment:
We could capture the just-built network and pass it to the `cloud_subnets` function so that we don't lazy_find it over and over again:

```ruby
network_obj = persister.cloud_networks.build(...)
cloud_subnets(network_obj, network.subnets) if network.subnets
```